orocos_kdl/tests/framestest.cpp

```diff
@@ -635,50 +635,63 @@ void FramesTest::TestFrame() {
 	CPPUNIT_ASSERT_EQUAL(F.Inverse()*v,F.Inverse(v));
 }
 
+JntArray CreateRandomJntArray(int size)
+{
+    JntArray j(size);
+    for (int i = 0; i<size; ++i)
+        random(j(i));
+    return j;
+}
+
 void FramesTest::TestJntArray()
 {
-    JntArray a1(4);
-    random(a1(0));
-    random(a1(1));
-    random(a1(2));
-    random(a1(3));
-    JntArray a2(a1);
-    CPPUNIT_ASSERT(Equal(a2,a1));
+    JntArray random = CreateRandomJntArray(4);
+    JntArray random_copy(random);
+    CPPUNIT_ASSERT(Equal(random_copy,random));
 
-    SetToZero(a2);
-    CPPUNIT_ASSERT(!Equal(a1,a2));
-    
-    JntArray a3(4);
-    CPPUNIT_ASSERT(Equal(a2,a3));
-    
-    a1=a2;
-    CPPUNIT_ASSERT(Equal(a1,a3));
+    JntArray zero_set_to_zero(4);
+    SetToZero(zero_set_to_zero);
+    CPPUNIT_ASSERT(!Equal(random,zero_set_to_zero));
+
+    JntArray zero(4);
+    CPPUNIT_ASSERT(Equal(zero_set_to_zero,zero));
+
+    JntArray almost_zero = CreateRandomJntArray(4);
+    almost_zero(0) = almost_zero(0)*1e-7;
+    almost_zero(1) = almost_zero(1)*1e-7;
+    almost_zero(2) = almost_zero(2)*1e-7;
+    almost_zero(3) = almost_zero(3)*1e-7;
+
+    // This should obviously be equal, but fails in old buggy implementation
+    CPPUNIT_ASSERT(Equal(almost_zero,zero,1));
+    CPPUNIT_ASSERT(Equal(almost_zero,zero,1e-6));
+    CPPUNIT_ASSERT(!Equal(almost_zero,zero,1e-8));
```

Comment:
There is a tiny chance that this assertion fails, if all random numbers generated above would be smaller than `1e-1` in magnitude. But I guess it is acceptable, and it is probably not the only case where randomness in test cases is potentially problematic.

Reply:
True, I have not experienced it yet. But we will see. I am also trying to fix other cases, where precision is not met because of randomness.